Add SCIM provisioning documentation for HCP (#2561)
hashicorp/web-unified-docs: content/hcp-docs/content/docs/hcp/iam/groups.mdx

To learn more about user permissions, refer to [User permissions](/hcp/docs/hcp/iam/users#user-permissions).

## Group provisioning methods

HCP supports two methods for creating and managing groups in your organization:

- **Manual provisioning** - Create groups directly in HCP and manually manage membership. This is the default method.
- **SCIM provisioning** - Automatically sync groups and their membership from your identity provider when SCIM is enabled. Groups are created, updated, and removed based on changes in your identity provider.

For more information about SCIM provisioning, refer to [SCIM provisioning](/hcp/docs/hcp/iam/sso/scim).

## Requirements

You must have [admin permissions for your organization](/hcp/docs/hcp/iam/users#user-permissions) to create and manage groups and users.

## Create a group

<Note>

> If SCIM provisioning is enabled for your organization, groups are automatically synchronized from your identity provider. Manual group creation is not available when SCIM is active. To manage groups, make changes in your identity provider. For more information, refer to [SCIM provisioning](/hcp/docs/hcp/iam/sso/scim).

Author: I am not sure this is true

> </Note>

Reviewer: ```suggestion
When SCIM provisioning is enabled for your organization, groups are automatically synchronized from your identity provider. You can manually create groups in HCP, but the SCIM-managed identity provider group takes precedence over an existing HCP group with the same name. As a result, the SCIM-managed identity provider group replaces the manually created HCP group. For more information, refer to [SCIM provisioning](/hcp/docs/hcp/iam/sso/scim).
```

1.  Search for and then select the groups you want to assign projects and roles to.
1.  Select a service and role to set the group's permissions for the project.
1.  Click **Save**.

## Manage SCIM-provisioned groups

When SCIM provisioning is enabled, group lifecycle management is handled by your identity provider:

- **Group creation** - Groups are automatically created in HCP when they exist in your identity provider
- **Group updates** - Group name changes in your identity provider automatically sync to HCP
- **Membership management** - Group membership is controlled in your identity provider and automatically synced to HCP
- **Group removal** - Groups deleted from your identity provider are automatically removed from HCP

> - **Role assignments** - You can still assign roles to SCIM-provisioned groups in HCP

Reviewer 2: ```suggestion
- **Group creation** - Groups are automatically created in HCP when they exist in your identity provider.
- **Group updates** - Group name changes in your identity provider automatically sync to HCP.
- **Membership management** - Group membership is controlled in your identity provider and automatically synced to HCP.
- **Group removal** - Groups deleted from your identity provider are automatically removed from HCP.
- **Role assignments** - You can still assign roles to SCIM-provisioned groups in HCP.
```

> - **Role assignments** - You can still assign roles to SCIM-provisioned groups in HCP.

Reviewer 3: You must assign roles to SCIM-provisioned groups.

> - **Role assignments** - You can still assign roles to SCIM-provisioned groups in HCP.

Reviewer: ```suggestion
- Group creation: Groups are automatically created in HCP when they exist in your identity provider.
- Group updates: Group name changes in your identity provider automatically sync to HCP.
- Membership management: Group membership is controlled in your identity provider and automatically synced to HCP.
- Group removal: Groups deleted from your identity provider are automatically removed from HCP.
- Role assignments: You must assign roles to SCIM-provisioned groups in HCP.
```

> To identify SCIM-provisioned groups, check the **SCIM provisioning** column in the groups list. Groups with a status of "Syncing" are managed by SCIM.

Reviewer 4: ```suggestion
To identify SCIM-provisioned groups, check the **SCIM provisioning** column in the groups list. Groups with a status of **Synced** are managed by SCIM.
```
